instrumentation-api-semconv/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/net/PeerServiceAttributesExtractor.java

```diff
@@ -53,27 +53,35 @@ public void onEnd(
       @Nullable RESPONSE response,
       @Nullable Throwable error) {
 
-    if (peerServiceMapping.isEmpty()) {
+    if (peerServiceResolver.isEmpty()) {
       // optimization for common case
       return;
     }
 
     String serverAddress = attributesGetter.getServerAddress(request);
-    String peerService = mapToPeerService(serverAddress);
+    Integer serverPort = attributesGetter.getServerPort(request);
+    // TODO wire protocol and path
+    String peerService = mapToPeerService(null, serverAddress, serverPort, null);
     if (peerService == null) {
       String serverSocketDomain = attributesGetter.getServerSocketDomain(request, response);
-      peerService = mapToPeerService(serverSocketDomain);
+      Integer serverSocketPort = attributesGetter.getServerSocketPort(request, response);
+      // TODO wire protocol and path
```

Comment:
Do you want to get this TODO in this PR? It looks really strange right now since all calls to `mapToPeerService()` pass a `null` protocol and `null` path.

Reply:
I would like it to be done in this PR but I’m not sure how to proceed, 
What would be the most appropriate way.? To my understanding the PeerServiceExtractor is only used for clients.
Shall I try to cast an httclient and parse the urlfull it provides for example ? But what about the other kind of clients ?

Reply:
I would like to be able to do something like
```java
String scheme = attributesGetter.getUrlScheme(request);
String path = attributesGetter.getUrlPath(request);
```

Reply:
Hi @trask , would have any idea on this part ?